CHANGELOG.md

```diff
@@ -90,6 +90,7 @@
   * `cortex_bucket_store_chunk_pool_returned_bytes_total`
 * [ENHANCEMENT] Alertmanager: load alertmanager configurations from object storage concurrently, and only load necessary configurations, speeding configuration synchronization process and executing fewer "GET object" operations to the storage when sharding is enabled. #3898
 * [ENHANCEMENT] Blocks storage: Ingester can now stream entire chunks instead of individual samples to the querier. At the moment this feature must be explicitly enabled either by using `-ingester.stream-chunks-when-using-blocks` flag or `ingester_stream_chunks_when_using_blocks` (boolean) field in runtime config file, but these configuration options are temporary and will be removed when feature is stable. #3889
+* [ENHANCEMENT] Alertmanager: cleanup obsolete local files after Alertmanager is no longer running for removed or resharded user. #3910
```

Comment:
I'd think this is a CHANGE, a user that deleted its config, waited a few minutes and re-created it would still have their silences configured.

Reply:
Right, although it is a change visible by end-user, but not for operator.